### ISSUE TITLE
Bug: Unions broken in python 3.10

### ISSUE THREAD

**Issue by llucid-97** [tamuhey/dataclass_utils#8]
Hi there, I ran into another bug while using this:

I'm using python 3.10 which has some syntax sugar for creating unions using the | operator from https://peps.python.org/pep-0604/

Interestingly, this causes the type checking to fail, when using the old way of creating a `typing.Union` object doesn't:

Here's the base case:

```python
import dataclasses
import dataclass_utils
import typing


@dataclasses.dataclass
class Foo:
    bar: typing.Union[str, int] = "bar"


f = Foo()
dataclass_utils.check_type(f)
print(f"Pass! {f=}")
```

This passes as expected:

> /home/agimg/anaconda3/envs/pbrl/bin/python /home/agimg/projects/fastpbrl/scripts/scratchpads/dataclass_utils_bug_2.py
> Pass! f=Foo(bar='bar')
> 
> Process finished with exit code 0

But using the syntax sugar:

```python
import dataclasses
import dataclass_utils


@dataclasses.dataclass
class FooWithSugar:
    bar: str | int = "bar"


f_sugar = FooWithSugar()
dataclass_utils.check_type(f_sugar)
print(f"Pass! {f_sugar=}")
```

Fails with:

> /home/agimg/anaconda3/envs/pbrl/bin/python /home/agimg/projects/fastpbrl/scripts/scratchpads/dataclass_utils_bug_2.py
> Traceback (most recent call last):
>   File "/home/agimg/projects/fastpbrl/scripts/scratchpads/dataclass_utils_bug_2.py", line 25, in <module>
>     dataclass_utils.check_type(f_sugar)
>   File "/home/agimg/anaconda3/envs/pbrl/lib/python3.10/site-packages/dataclass_utils/type_checker.py", line 189, in check_root
>     raise err
> dataclass_utils.error.Error0: Error in field 'bar'. Expected type <class 'types.UnionType'>, got <class 'str'> (value: bar)
> 
> Process finished with exit code 1

Do you plan on supporting this new syntax?

**Comment by tamuhey** [comment]
Hi, thanks for reporting the bug! I will try to create a patch.

**Comment by tamuhey** [comment]
Just published a new version 0.7.22, which should solve this issue.

**Comment by llucid-97** [comment]
It works for me now. Thanks
